frontend/ui/widget/bookmapwidget.lua

```diff
@@ -1027,6 +1027,7 @@ On a newly opened book, the book map will start in grid mode showing all chapter
 end
 
 function BookMapWidget:onClose(close_all_parents)
+    BD.resetinvert()
```

Comment:
This is better be done in the else branch of the `if self.launcher then`, where the last of BookMapWidget or PageBrowserWidget is closed. (You can launch BMW from PBW or PBW from BMW, so have a stack of many of them - we don't want to most recent one to mess the state of things if it has to re-make some UI layout and create new widgets who would get this flag reset.)

Also, better BD.invert() in one of these widget:init() instead of ReaderThumbnail:onShowBookMap() when there is no `self.launcher`. For now, only ReaderThumbnail create them, but may be some day, we'll use these widgets in other context (like from ReaderSearch to show results in a BookMapWidget to see where search word occurs in the book).

Reply:
I wasn't sure where to put then and didn't realy finish with the bookmap part as i saw it didn't work so i was waiting on that